Issue/v4/1027 delete trash

stefanorosanelli opened the pull request:
This PR solves #1027 

Conditions on `deleted` field are set in Controllers: I know there are better ways to do that :smiley_cat:

stefanorosanelli committed:
#1027 feat: new objects.deleted field

stefanorosanelli committed:
#1027 feat: soft delete, hide objects with deleted = 1

stefanorosanelli committed:
#1027 feat: 'deleted' validation rules

stefanorosanelli committed:
#1027 fix: use 0 instead of false  on sqlite/pg

batopa commented on plugins/BEdita/API/src/Controller/UsersController.php:
```diff
@@ -153,7 +157,9 @@ public function delete($id)
+        $user = $this->Users->get($id, [
+            'conditions' => ['deleted' => 0]
+        ]);
         if (!$this->Users->delete($user)) {
```
The user is really deleted. Add unit test on user delete case

stefanorosanelli replied:
Right.
Solved and modified unit test.
I had to use `ObjectsTable` instead of `UserTable` in `UsersController::delete()` - I'm opening a new issue for that

stefanorosanelli committed:
#1027 fix: use `deleted` field on user delete

NOTE: $this->Users->save() not working as expected, to be solved in other issue

batopa commented:
Ok, I would merge it and open a new issue to better handle soft delete. 

I think it's better to use a behavior to have consistency every time a save/find on an object is done. Moreover the behavior could be expose some useful custom finders for example to get just trashed objects.
